posthog/management/commands/verify_team_metadata_cache.py

```diff
@@ -0,0 +1,240 @@
+"""
+Management command to verify team metadata cache consistency.
+
+Compares cached data against database to detect discrepancies.
+
+Usage:
+    # Verify all teams
+    python manage.py verify_team_metadata_cache
+
+    # Verify specific teams
+    python manage.py verify_team_metadata_cache --team-ids 123 456 789
+
+    # Sample random teams
+    python manage.py verify_team_metadata_cache --sample 100
+
+    # Check dedicated flags cache too
+    python manage.py verify_team_metadata_cache --check-dedicated-cache
+
+    # Verbose output (show full diffs)
+    python manage.py verify_team_metadata_cache --verbose
+"""
+
+from django.conf import settings
+from django.core.cache import caches
+from django.core.management.base import BaseCommand
+
+from posthog.caching.flags_redis_cache import FLAGS_DEDICATED_CACHE_ALIAS
+from posthog.models.team.team import Team
+from posthog.storage.team_metadata_cache import (
+    TEAM_METADATA_FIELDS,
+    _serialize_team_field,
+    get_team_metadata,
+    team_metadata_hypercache,
+)
+
+
+class Command(BaseCommand):
+    help = "Verify team metadata cache consistency against database"
+
```

Comment:
Should we add a `--fix` option to automatically fix the teams we find?

Reply:
good idea